lib/src/connection.dart

```diff
@@ -380,15 +380,19 @@ abstract class _PostgreSQLExecutionContextMixin implements PostgreSQLExecutionCo
 
   Future _resolveTableOIDs(List<int> oids) async {
     final unresolvedIDString = oids.join(",");
-    final orderedTableNames =
-        await query("SELECT relname FROM pg_class WHERE relkind='r' AND oid IN ($unresolvedIDString) ORDER BY oid ASC");
-
-    final iterator = oids.iterator;
-    orderedTableNames.forEach((tableName) {
-      iterator.moveNext();
-      if (tableName.first != null) {
-        _tableOIDNameMap[iterator.current] = tableName.first;
-      }
+    final orderedTableNames = await query(
+        "SELECT oid, relname FROM pg_class WHERE relkind='r' AND oid IN ($unresolvedIDString) ORDER BY oid ASC");
+
+    final unresolvedOids = oids.toSet();
```

Comment:
Does it make sense to move this line to the top of this function and use it when preparing `unresolvedIDString`?